onedal/_device_offload.py

```diff
@@ -14,28 +14,137 @@
 # limitations under the License.
 # ==============================================================================
 
+import logging
+from collections.abc import Iterable
 from functools import wraps
 
+import numpy as np
+
+from ._config import _get_config
+
 try:
-    import dpnp
+    from dpctl import SyclQueue
+    from dpctl.memory import MemoryUSMDevice, as_usm_memory
+    from dpctl.tensor import usm_ndarray
 
-    dpnp_available = True
+    dpctl_available = True
 except ImportError:
-    dpnp_available = False
+    dpctl_available = False
 
 try:
-    from sklearnex._device_offload import (
-        _copy_to_usm,
-        _get_global_queue,
-        _transfer_to_host,
-    )
+    import dpnp
 
-    _sklearnex_available = True
+    dpnp_available = True
 except ImportError:
-    import logging
+    dpnp_available = False
+
 
-    logging.warning("Device support requires " "Intel(R) Extension for Scikit-learn*.")
-    _sklearnex_available = False
+class DummySyclQueue:
+    """This class is designed to act like dpctl.SyclQueue
+    to allow device dispatching in scenarios when dpctl is not available"""
+
+    class DummySyclDevice:
+        def __init__(self, filter_string):
+            self._filter_string = filter_string
+            self.is_cpu = "cpu" in filter_string
+            self.is_gpu = "gpu" in filter_string
+            # TODO: check for possibility of fp64 support
```

Comment:
It is necessary when we are trying to offload GPU via config_context, when dpctl is not in env.
I will cover this with JIRA ticket. I was not thinking to change a lot in the offloading flow. Make sense to work on it seperatly.

Reply:
Covered with ticket. Will remove TODO comment from the source after last CI check completed.

Reply:
> DummySyclQueue and why its necessary to set has_aspect_fp64 to false. Wouldn't a warning or throwing an error be better? I know this a copy of sklearnex/_device_offload,py, but it may be a good time to address this.

I don't think that this is a good time to address. This should be investigated and adressed separately. I don't want it be a blocker. This is just copy of original impl. TODO comment already covered by related ticket, that should be prioritized and adressed.